src/network/core/address.cpp

```diff
@@ -24,8 +24,8 @@
 const char *NetworkAddress::GetHostname()
 {
 	if (StrEmpty(this->hostname) && this->address.ss_family != AF_UNSPEC) {
-		assert(this->address_length != 0);
-		getnameinfo((struct sockaddr *)&this->address, this->address_length, this->hostname, sizeof(this->hostname), nullptr, 0, NI_NUMERICHOST);
+		int size = this->address.ss_family == AF_INET6 ? sizeof(struct sockaddr_in6) : sizeof(struct sockaddr_in);
```

Comment:
why this change? seems like it should be explained sufficiently and split into its own commit
(and this->address_length removed?)

Reply:
Emscripten getnameinfo returns error when passed size other than sizeof(sockaddr_in)
https://github.com/emscripten-core/emscripten/issues/7636
It isn't guarded by emscripten specific ifdef as it shouldn't hurt other platforms.